states/huntsman/twilight_flat_fielding.py

```diff
@@ -0,0 +1,71 @@
+"""
+State to handle the taking of calibration frames (evening and morning).
+"""
+
+
+def _get_cameras(pocs):
+    '''
+    Get lists of narrow and broad band cameras.
+    '''
+    narrow_band_cameras = list()
+    broad_band_cameras = list()
+    for cam_name, cam in pocs.observatory.cameras.items():
+
+        # This is a hack. There should be a narrow property in the config...
+        if cam.filter_type.lower().startswith('ha'):
+            narrow_band_cameras.append(cam_name)
+        else:
+            broad_band_cameras.append(cam_name)
+
+    return narrow_band_cameras, broad_band_cameras
+
+
+def _wait_for_twilight(pocs):
+    '''
+    Wait for twilight. Temporary solution until something better is found.
+    '''
+    pocs.logger.debug('Waiting for twilight...')
+    while pocs.is_dark(horizon='focus') or not pocs.is_safe(horizon='flat'):
+        pocs.sleep(delay=pocs._safe_delay)
+
+
+def on_enter(event_data):
+    '''
+    Calibrating state. If safe to do so, take flats and darks. Should be
+    called once at the beginning and end of the night.
+
+    If evening, the next state will be coarse_focusing, else, parking.
+    '''
+    pocs = event_data.model
+    pocs.next_state = 'parking'
+
+    # Make sure its safe, dark and light enough for flats
+    _wait_for_twilight(pocs)
+
+    if pocs.observatory.take_flat_fields:
```

Comment:
Do we still even want this conditional? In what operational situations do we not want to take flats?

Reply:
For Huntsman we'll always want to take them when we can, and since this is nominally Huntsman specific code then I guess it's not necessary.

Reply:
Not necessary, no. But I've found it useful for unit testing the state machine without running all the code. Note that the `take_morning_flats` function is not yet implemented so the state machine tests would throw and error without this functionality. Its not perfect, but its not a problem for this PR to solve.